lib/Sema/DerivedConformanceDifferentiable.cpp

```diff
@@ -736,6 +736,11 @@ getOrSynthesizeSingleAssociatedStruct(DerivedConformance &derived,
     // call to the getter.
     if (member->getEffectiveAccess() > AccessLevel::Internal &&
         !member->getAttrs().hasAttribute<DifferentiableAttr>()) {
+      // If member or its getter already has a `@differentiable` attribute,
+      // continue.
+      if (member->getAttrs().hasAttribute<DifferentiableAttr>() ||
+          member->getGetter()->getAttrs().hasAttribute<DifferentiableAttr>())
```

Comment:
I don't think you need to check the getter decl because users can never write a getter for a stored property.

Reply:
Actually, `TypeCheckAttr::visitDifferentiableAttr` transfers `@differentiable` attributes from `AbstractStorageDecl` declarations (including stored properties) to their getters.

The `member->getGetter()->getAttrs().hasAttribute<DifferentiableAttr>()` check is necessary; without it, there's a duplicate attribute assertion violation in SILFunctionBuilder.cpp. I also kept the `member->getAttrs().hasAttribute<DifferentiableAttr>()` check for robustness.